app/controllers/urls_controller.rb

```diff
@@ -0,0 +1,55 @@
+class UrlsController < ApplicationController
+  before_action :find_url, only: %i[show update]
+
+  def index
+    urls = Url.all.order("pinned DESC, updated_at DESC")
+    render status: :ok, json: { urls: urls }
+  end
+
+  def create
+    url = Url.new(url_params)
+    url.slug = generate_slug
```

Comment:
Controller's only job is to receive a request, fetch data from the db and send it to view.

All other business logic should go in the model, like generate_slug